src/App.js

```diff
@@ -43,19 +48,38 @@ function App() {
     console.log(cartProducts)
   };
 
-  const onAddToFavorite = (obj) => {
-    setFavorites(prev => {
-      // Check if the item already exists in favorites
-      if (!prev.some(item => item.id === obj.id)) {
-        // If it doesn't exist, add it to the favorites array
-        return [...prev, obj];
+  const onAddToFavorite = async (obj) => {
+    console.log(obj)
```

Comment:
remove this redundant logging

If you want to keep it, make it more readable, e.g. console.log("Adding to favourite: ", obj)

Reply:
ok